Core/Utilities/ElasticSearchDB.py

```diff
@@ -0,0 +1,272 @@
+Elasticsearch database.
+
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC                      import gLogger, S_OK, S_ERROR
+from elasticsearch              import Elasticsearch
+from elasticsearch_dsl          import Search, Q, A
+from elasticsearch.exceptions   import ConnectionError, TransportError, NotFoundError
+from elasticsearch.helpers      import BulkIndexError
+from elasticsearch              import helpers
+from datetime                   import datetime
+from DIRAC.Core.Utilities       import Time
+
+class ElasticSearchDB( object ):
+  
+  """
+  .. class:: ElasticSearchDB
+
+  :param str url: the url to the database for example: el.cern.ch:9200
+  :param str gDebugFile: is used to save the debug information to a file
+  """
+  __chunk_size = 1000
+  __url = ""
+  ########################################################################
+  def __init__( self, host, port, debug = False ):
+    """ c'tor
+    :param self: self reference
+    :param str host: name of the database for example: MonitoringDB
+    :param str port: The full name of the database for example: 'Monitoring/MonitoringDB'
+    :param bool debug: save the debug information to a file   
+    """
+    global gDebugFile
+    
+    if 'log' not in dir( self ):
+      self.log = gLogger.getSubLogger( 'ElasticSearch' )
+    self.logger = self.log
+    
+    self.__url = "http://%s:%s" % ( host, port )
+        
+    if debug:
+      try:
+        gDebugFile = open( "%s.debug.log" % self.__dbName, "w" )
+      except IOError as e:
+        self.log.error( e )
+      
+    self.__client = Elasticsearch( self.__url )
+    self.__tryToConnect()
+  
+  ########################################################################
+  def setClusterName( self, name ):
+    """
+      It is used to set the cluster name
+      
+      :param self: self reference
+      :param str requestName: request name
+    
+    """
+    self.__clusterName = name
+  
+  ########################################################################
+  def getClusterName( self ):
+    """
+    It returns the cluster name
+    
+    :param self: self reference
+    """
+    return self.__clusterName
+  
+  ########################################################################  
+  def query( self, index, query ):
+    """It exexutes a query and it returns the result
+    query is a dictionary. More info: search for elasticsearch dsl
+    
+    :param self: self reference
+    :param dict query: It is the query in ElasticSerach DSL language
+     
+    """
+    return self.__client.search( index = index, body = query )
+  
+  def _Search( self, indexname ):
+    """
+    it returns the object which can be used for reatriving ceratin value from the DB
+    """
+    return  Search( using = self.__client, index = indexname )
+  
+  ########################################################################
+  def _Q( self, name_or_query = 'match', **params ):
```

Comment:
As you can see this is an elasticsearch module wrapper. I want to have a single import of this elasticserach modules.